src/sim/kernel_workload.cc

```diff
@@ -102,12 +102,29 @@ KernelWorkload::initState()
     auto mapper = [this](Addr a) {
         return (a & _loadAddrMask) + _loadAddrOffset;
     };
+    auto memory_size = system->memSize();
     if (params().object_file != "")  {
         if (params().addr_check) {
+            // Check that memory size is large enough to contain the kernel.
+            if (memory_size < _end){
+                inform("You currently do not have enough memory in your "
+                "configuration to accomodate the size of the kernel.\n");
+                inform("End of kernel location in memory: %#x\n", _end);
+                inform("Size of memory: %#x = %dMB\n", memory_size,
+                memory_size>>20);
+                if ((_end & 0xFFFFF) != 0){
+                    inform("Increase your memory size to at least %dMB to fit"
+                    " the kernel.\n", (_end>>20) + 1);
+                }
+                else{
+                    inform("Increase your memory size to at least %dMB to fit"
+                    " the kernel.\n", _end>>20);
+                }
+            }
             // Validate kernel mapping before loading binary
             fatal_if(!system->isMemAddr(mapper(_start)) ||
```

Comment:
I was the one of actually asked @erin-le to do this. My main complaint with the current error is while it correctly reports the error, a subset of this error is undoubtably the case where there simply isn't enough memory to load the kernel and in this case it'd be useful to have this communicated directly. It's a simple mistake for a beginner to make and the error returned might be hard for some to fully understand. I think determining this specific bug and specializing the error returned is easily doable and worthwhile.

Would you prefer if the more specific error was appended to the "Kernel is mapped to invalid location"? I've suggested this in my other comment. It could keep things more concise.